overlord/assertstate/assertstate_test.go

```diff
@@ -1142,3 +1143,60 @@ func (s *assertMgrSuite) TestPublisher(c *C) {
 	c.Check(acct.AccountID(), Equals, s.dev1Acct.AccountID())
 	c.Check(acct.Username(), Equals, "developer1")
 }
+
+func (s *assertMgrSuite) TestAutoRefresh(c *C) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	snapDeclFoo := s.snapDecl(c, "foo", nil)
+	s.stateFromDecl(snapDeclFoo, snap.R(7))
+
+	// previous state
+	err := assertstate.Add(s.state, s.storeSigning.StoreAccountKey(""))
+	c.Assert(err, IsNil)
+	err = assertstate.Add(s.state, s.dev1Acct)
+	c.Assert(err, IsNil)
+	err = assertstate.Add(s.state, snapDeclFoo)
+	c.Assert(err, IsNil)
+	// also another account-key
+	err = assertstate.Add(s.state, s.dev1AcctKey)
+	c.Assert(err, IsNil)
+
+	// one changed snap-decl
+	headers := map[string]interface{}{
+		"series":       "16",
+		"snap-id":      "foo-id",
+		"snap-name":    "fo-o",
```

Comment:
I think it would have been easier to follow if this was just `new` and `old` above.

Unless I misunderstand and something else is being tested but then my point stands more :)